include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -158,7 +158,12 @@ struct __subgroup_radix_sort
                 auto __counter_lacc = __buf_count.get_acc(__cgh);
 
                 __cgh.parallel_for<_Name...>(
-                    __range, ([=](sycl::nd_item<1> __it)[[_ONEDPL_SYCL_REQD_SUB_GROUP_SIZE(__req_sub_group_size)]] {
+                    __range, ([=](sycl::nd_item<1> __it)
+                        // CUDA devices only support sg size of 32, so we disable this requirement
```

Comment:
Would we need to guard this for other backends as well? AMD seems to support 64 and 32 depending on their configuration (https://github.com/RadeonOpenCompute/ROCm/discussions/2318).

Reply:
Yes, if there is a reliable way to determine if we are compiling for the HIP backend, then we should guard against this as well. I will take a look to see if we can do this.